Updated Jenkinsfile for master branch release strategy

awharris opened the pull request:
- Refactored to use only master branch
- Currently only deploys into dev environment

Commit:
Updated Jenkinsfile for master branch release strategy

- Refactored to use only master branch
- Currently only deploys into dev environment

Commit:
Fixed path to Cobertura report

- Corrected path Cobertura report

Commit:
Added Scala Style config

- Added config file for Scala Style static analysis
- Added manifest.yml to .gitignore

Commit:
Fixed typo in jenkinsfile

- Missing quote terminator added

Commit:
Temp removal of master branch check

- Removed master branch check to validate package and deploy steps

Commit:
Fixed package step

- Corrected path when pulling from Gitlab
- Fixed "deploy to" environment variable
- Corrected target app name

Commit:
Fixed deploy step

- Fixed gitlab checkout overriding play properties file

Commit:
Fixed checkstyle report publishing

- Added path to generate scapegoat reports in Checkstyle publisher
- Reinstated master branch check for deployment

willis7 commented on Jenkinsfile:
```diff
@@ -1,25 +1,19 @@
-#!groovy
+#!groovyF
```
Is this a typo? I cant find any reference to `groovyF`

awharris replied:
Typo

willis7 commented on Jenkinsfile:
```diff
@@ -29,225 +23,154 @@ pipeline {
-                sh "$SBT version"
+                sh "sbt version"
                 script {
                     version = '1.0.' + env.BUILD_NUMBER
```
I'm not sure how useful this is. Because it's hardcoded and not taken from sbt they could fall out of sync. I would either remove and just use the build number, or we devise a solution for printing sbt properties (which I have tried, but wasnt successful with).

awharris replied:
sure, removed hardcoded version prefix

willis7 commented on Jenkinsfile:
```diff
@@ -29,225 +23,154 @@ pipeline {
             steps {
-                script {
-                    env.NODE_STAGE = "Bundle"
+                parallel (
```
This is the scripted style syntax. If you want to use declarative then see here: 
https://jenkins.io/doc/book/pipeline/syntax/#parallel

awharris replied:
Done

willis7 commented on Jenkinsfile:
```diff
@@ -29,225 +23,154 @@ pipeline {
-                    //push(newTag, currentTag)
+                dir('gitlab') {
+                    git(url: "$GITLAB_URL/BusinessIndex/${MODULE_NAME}.git", credentialsId: GITLAB_CREDS, branch: "master")
                 }
```
The git clone isnt required yet. This can be moved to deploy.

awharris replied:
Moved

willis7 commented on Jenkinsfile:
```diff
@@ -29,225 +23,154 @@ pipeline {
+                    git(url: "$GITLAB_URL/BusinessIndex/${MODULE_NAME}.git", credentialsId: GITLAB_CREDS, branch: "master")
                 }
+                sh 'sbt universal:packageBin'
+                sh "cp target/universal/${ORGANIZATION}-${MODULE_NAME}-*.zip ${DEPLOY_TO}-${ORGANIZATION}-${MODULE_NAME}.zip"
```
I personally dont see the point of renaming the file here

willis7 commented on Jenkinsfile:
```diff
@@ -29,225 +23,154 @@ pipeline {
                 milestone(1)
-                lock('Deployment Initiated') {
-                    colourText("info", 'deployment in progress')
+                lock('Business Index API Deployment Initiated') {
```
We should pick a naming scheme for locks so it's clear a resource is being locked (it's not a description per-se).

It might make sense here to use the CF route as the lock id.

willis7 reviewed (commented):
Many of the points I made were from my own todo list. I put them here too for transparency.

Commit:
Feedback from review

- Changed lock name to CF route
- Corrected typo
- Changed version number to build number
- Moved gitlab clone to deploy stage
- Updated SBT to create package with standard name
- Replaced scripted parallel step with declarative version

Commit:
Now uses STAGE_NAME env variable

- Uses inbuilt STAGE_NAME env variable
- Switched code coverage off after report generated
- Moved post step after each stage in parallel execution

Commit:
Moved parallel step posts to end of stage

- Seems post is only valid after all parallel steps

willis7 commented:
Here's the declarative way to add libraries in Jenkins: https://github.com/ONSdigital/sbr-control-api/blob/feature/REG-1052-continuous-delivery/Jenkinsfile#L7

Commit:
Merge branch 'master' into feature/REG-1417

Commit:
Merge branch 'master' into feature/REG-1417
